### PR TITLE
tests: Add fuzzing harness for functions/classes in flatfile.h, merkleblock.h, random.h, serialize.h and span.h

### PR DESCRIPTION
**Pull request by practicalswift**
* Add fuzzing harness for functions/classes in `flatfile.h`
* Add fuzzing harness for functions/classes in `merkleblock.h`
* Add fuzzing harness for functions/classes in `span.h`
* Add fuzzing harness for `LimitedString` (`serialize.h`)
* Add fuzzing harness for functions/classes in `random.h`

### CODE REVIEW

File: src/test/fuzz/flatfile.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+    Optional<FlatFilePos> flat_file_pos = ConsumeDeserializable<FlatFilePos>(fuzzed_data_provider);
+    if (!flat_file_pos) {
+        return;
+    }
+    Optional<FlatFilePos> another_flat_file_pos = ConsumeDeserializable<FlatFilePos>(fuzzed_data_provider);
+    if (another_flat_file_pos) {
+        (void)(*flat_file_pos == *another_flat_file_pos);
+        (void)(*flat_file_pos != *another_flat_file_pos);
```
Comment (MarcoFalke) [inline src/test/fuzz/flatfile.cpp]:
```suggestion
        assert((*flat_file_pos == *another_flat_file_pos) != (*flat_file_pos != *another_flat_file_pos));
```

File: src/test/fuzz/span.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        (void)span.last(idx);
+        (void)span.subspan(idx);
+        (void)span.subspan(idx, span.size() - idx);
+        (void)span[idx];
+    }
+
+    std::string another_str = fuzzed_data_provider.ConsumeBytesAsString(32);
+    const Span<const char> another_span = MakeSpan(another_str);
+    (void)(span == another_span);
+    (void)(span != another_span);
```
Comment (MarcoFalke) [inline src/test/fuzz/span.cpp]: Same here

File: src/test/fuzz/span.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    (void)(span < another_span);
+    (void)(span <= another_span);
+    (void)(span > another_span);
+    (void)(span >= another_span);
```
Comment (MarcoFalke) [inline src/test/fuzz/span.cpp]:
```suggestion
    assert((span >= another_span) != (span < another_span));
```

**Review by maflcko — APPROVED** [review]

File: src/test/fuzz/flatfile.cpp (hunk 1 above)
Comment (practicalswift) [reply src/test/fuzz/flatfile.cpp]: Good idea! Fixed!

**Comment by practicalswift** [comment]
@MarcoFalke Thanks for nice suggestions. Now implemented. Please re-review :)

**Pull request merged**